tests/test_bgp_af.rb

```diff
@@ -152,51 +206,77 @@ def test_default_information_originate
   ## client-to-client reflection
   ##
   def test_client_to_client
-    asn = '55'
-    vrf = 'red'
-    af = %w(ipv4 unicast)
-
-    bgp_af = RouterBgpAF.new(asn, vrf, af)
-    pattern = /^ *client-to-client reflection$/
-    #
-    # Default is 'client-to-client' is configured
-    #
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-
-    assert_match(pattern, af_string,
+    if platform == :ios_xr
+      # default is client-to-client reflection enabled but doesn't show up in config
+      # router bgp 55
+      #  address-family ipv4 unicast
+      #   bgp client-to-client reflection disable
+      asn = '55'
+      vrf = 'default'
+      af = %w(ipv4 unicast)
+      config_ios_xr_dependencies(asn)
+      bgp_af = RouterBgpAF.new(asn, vrf, af)
+      af_string = @device.cmd("show run router bgp | in bgp client-to-client")
+      pattern = /^ *bgp client-to-client reflection disable$/
+      refute_match(pattern, af_string, "bgp client-to-client disable is configured but should not be")
+      # Set and verify
+      bgp_af.client_to_client = true
+      af_string = @device.cmd("show run router bgp | in bgp client-to-client")
+      assert_match(pattern, af_string, "bgp client-to-client disable is configured and should be")
+      # Unset and verify
+      bgp_af.client_to_client = false
+      af_string = @device.cmd("show run router bgp | in bgp client-to-client")
+      refute_match(pattern, af_string, "bgp client-to-client disable is configured but should not be")
+    elsif platform == :nexus
```

Comment:
This seems wrong - I would expect you'd be able to share most of the test logic between the platforms. I'm also concerned that on Nexus you're testing the getter (`assert(bgp_af.client_to_client)`) but on XR it doesn't look like you are?